### DISCUSSION TITLE
Single Object Multiple parts with Multiple Colours

### DISCUSSION THREAD

**Discussion by sheminasalam** [OrcaSlicer/OrcaSlicer#2438] [General]
Hi, I have made a number of 3D models with multiple parts in single 3mf file. I mostly use Fusion 360 for models and during modeling I assign colours to each part. But when I upload this 3mf file to the slicer, It assigns the primary filament colour to all the parts. I was wondering if the slicer would be able to distinguish the parts with different colours and assign either the original colour to the parts or even randon colours(which can be changed later). 

For example if my 3mf file has 5 colour parts, the slicer should be to detect them. Then it should find which colour has the maximum spread by area and assign the filament 1 to it. The second largest should be assigned the second filament and so on till the last. 

Will this be possible?

Also there should be an option to segregate parts according to colours on different filament beds and arrange them to avoid using transition towers for multi filament printing.

**Comment by JelbertHoltrop** [comment]
Yes I would like something like this too. I make mechanical components and would like to be able to use different filaments for different parts inside the object.
